crates/ruff_linter/src/noqa.rs

```diff
@@ -82,11 +82,19 @@ impl<'a> Directive<'a> {
 
                     // Extract the comma-separated list of codes.
                     let mut codes = vec![];
+                    let mut code_ranges = vec![];
                     let mut codes_end = codes_start;
                     let mut leading_space = 0;
                     while let Some(code) = Self::lex_code(&text[codes_end + leading_space..]) {
                         codes.push(code);
                         codes_end += leading_space;
+                        code_ranges.push(
+                            TextRange::new(
+                                TextSize::try_from(codes_end).unwrap(),
+                                TextSize::try_from(codes_end + code.len()).unwrap(),
+                            )
```

Comment:
Nit: you can use `at` to avoid having to offset the end by `codes_end`
```suggestion
                            TextRange::at(
                                TextSize::try_from(codes_end).unwrap(),
                                code.text_len(),
                            )
```

Reply:
Nice